### PR TITLE
capdl: refactor tcb_slots

### PR DESCRIPTION
**Pull request by corlewis**
This updates the TCB cap slots in the CapDL specification to match the output of the CapDL-tool. It also provides a new definition to hide these details and improve maintainability.

### CODE REVIEW

File: spec/capDL/Types_D.thy [hunk 1]
```diff
@@ -283,14 +283,21 @@ definition
 (*
  * Each TCB contains a number of cap slots, each with a specific
  * purpose. These constants define the purpose of each slot.
+ *
+ * The specific choice of slots is chosen to be consistent with the output
+ * of the CapDL-tool.
```
Comment (lsf37) [inline spec/capDL/Types_D.thy:288]:
```suggestion
 * The specific list of slots is chosen to be consistent with the output
 * of the CapDL-tool.
```

**Review by lsf37 — APPROVED** [review]
Nice. That will also make it easier to adjust those slots in the future.

**Comment by lsf37** [comment]
@corlewis can this be merged or is there something blocking it?

**Comment by corlewis** [comment]
> @corlewis can this be merged or is there something blocking it?

Nothing that I can remember, there was just that question about whether this was the best way or if the CapDL-tool should be adjusted instead. Like you said in the other PR, changes to these slots are infrequent enough that this solution should be fine.

**Pull request merged**